src/timessquare/services/page.py

```diff
@@ -64,18 +82,165 @@ async def render_page_template(
+                nbhtml = NbHtmlModel.create_from_noteburst_result(
+                    page_name=page.name,
+                    html=html,
+                    parameters=resolved_parameters,
+                    noteburst_result=noteburst_response,
+                )
+                # FIXME make lifetime a setting of page for pages that aren't
+                # idempotent.
+                await self._html_store.store(nbhtml=nbhtml, lifetime=None)
+                self._logger.debug("Stored new HTML")
+                await self._job_store.delete(
+                    page_name=page.name, parameters=resolved_parameters
+                )
+                self._logger.debug("Deleted old job record")
+                return nbhtml
+
+            else:
+                # Noteburst job isn't complete
+                return None
+
+        elif r.status_code != 404:
+            # Noteburst lost the job; delete our record and try again
+            self._logger.warning(
+                "Got a 404 from a noteburst job", job_url=job.job_url
+            )
+            await self._job_store.delete(
+                page_name=page.name, parameters=resolved_parameters
+            )
+            await self._request_noteburst_execution(
+                page=page, resolved_parameters=resolved_parameters
+            )
+        else:
+            # server error from noteburst
+            self._logger.warning(
+                "Got unknown response from noteburst job",
+                job_url=job.job_url,
+                noteburst_status=r.status_code,
+                noteburst_body=r.text,
+            )
+        return None
+
+    async def _request_noteburst_execution(
+        self, *, page: PageModel, resolved_parameters: Mapping[str, Any]
+    ) -> None:
+        """Request a notebook execution for a given page and parameters,
+        and store the job.
+        """
+        ipynb = page.render_parameters(resolved_parameters)
+        r = await self._http_client.post(
+            f"{config.environment_url}/noteburst/v1/notebooks/",
+            json={
+                "ipynb": ipynb,
+                "kernel_name": "LSST",  # TODO make a setting per page?
+            },
+            headers=self._noteburst_auth_header,
+        )
+        if r.status_code != 202:
+            self._logger.warning(
+                "Error requesting noteburst execution",
+                noteburst_status=r.status_code,
+                noteburst_body=r.text,
+            )
+
+            r2 = await self._http_client.get(
+                f"{config.environment_url}/noteburst/",
+                headers=self._noteburst_auth_header,
+            )
+            self._logger.warning(
+                "Pinged noteburst metadata",
+                noteburst_status=r2.status_code,
+                noteburst_body=r2.text,
+                noteburst_req_url=r2.request.url,
+                noteburst_req_headers=r2.request.headers,
+            )
```

Comment:
Not sure what this is: left-over debugging? Diagnostics? It feels like it's part of a health check endpoint.